File: frontend/src/features/administrative/service/administrativeArea.service.ts
import { ApiService } from "@/shared/services/api.service";
import {
    AdministrativeAreaResponse,
    AdministrativeAreaCreateRequest,
    AdministrativeAreaUpdateRequest,
} from "../type/administrativeAreaTypes";

export const AdministrativeAreasService = {
    getAllAreas: async (): Promise<AdministrativeAreaResponse[]> => {
        const response = await ApiService.get<AdministrativeAreaResponse[]>("/administrative-areas");
        return response.data;
    },

    getAreaById: async (id: string): Promise<AdministrativeAreaResponse> => {
        const response = await ApiService.get<AdministrativeAreaResponse>(`/administrative-areas/${id}`);
        return response.data;
    },

    createArea: async (
        request: AdministrativeAreaCreateRequest
    ): Promise<AdministrativeAreaResponse> => {
        const response = await ApiService.post<AdministrativeAreaResponse>(
            "/administrative-areas",
            request
        );
        return response.data;
    },

    updateArea: async (
        id: string,
        request: AdministrativeAreaUpdateRequest
    ): Promise<AdministrativeAreaResponse> => {
        const response = await ApiService.put<AdministrativeAreaResponse>(
            `/administrative-areas/${id}`,
            request
        );
        return response.data;
    },

    deleteArea: async (id: string): Promise<void> => {
        await ApiService.delete(`/administrative-areas/${id}`);
    },
};

File: backend/src/main/java/com/veterinaria/backend/administrative/service/AdministrativeAreaService.java
package com.veterinaria.backend.administrative.service;

import com.veterinaria.backend.administrative.dto.AdministrativeAreaDTO;
import com.veterinaria.backend.administrative.dto.CreateAdministrativeAreaDTO;
import com.veterinaria.backend.administrative.dto.UpdateAdministrativeAreaDTO;

import java.util.List;
import java.util.UUID;

public interface AdministrativeAreaService {
    List<AdministrativeAreaDTO> getAllAreas();
    AdministrativeAreaDTO getAreaById(UUID id);
    AdministrativeAreaDTO createArea(CreateAdministrativeAreaDTO dto);
    AdministrativeAreaDTO updateArea(UUID id, UpdateAdministrativeAreaDTO dto);
    void deleteArea(UUID id);
}
